ci_framework/roles/env_op_images/tasks/main.yml

```diff
@@ -0,0 +1,111 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure directory is present
+  ansible.builtin.file:
+    path: "{{ cifmw_env_op_images_dir }}/{{ item }}"
+    state: directory
+  loop:
+    - artifacts
+    - logs
+
+- name: Check if OpenStackControlPlane is setup
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell: |
+    oc get OpenStackControlPlane -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} --no-headers=true | awk '{print $2}'
+  register: openstack_installed
+
+- name: Get openstack operator images
+  when:
+    - openstack_installed.stdout == "True"
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ignore_errors: true
+  block:
+    - name: Create a file to store the images
+      ansible.builtin.copy:
+        dest: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file  }}"
+        content: "cifmw_install_yamls_vars:"
+
+    - name: Get all the pods in openstack-operator namespace
+      kubernetes.core.k8s_info:
+        kind: Pod
+        namespace: "{{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }}"
+        field_selectors:
+          - status.phase=Running
+      register: pod_list
+
+    - name: Retreive openstack-operator-index pod
+      set_fact:
+        selected_pod: "{{ pod_list.resources| selectattr('metadata.generateName', 'equalto', 'openstack-operator-index-') | list|first}}"
+
+    - name: Write openstack-operator-index image to file
+      ansible.builtin.lineinfile:
```

Comment:
Is this generating a yaml file? Why not using proper Ansible? This is hacky, I mean, there are enough builtin tools to do this. The thing that I expected here is a fact, created with set_fact and where you can add, by using the combine filter, every key-value with each image, and, at the very end, dump that fact to a file.

Reply:
Since openstack-operator-index and rabbitmq images require different image extraction, I kept the code as is. Changed it for other operators

Reply:
For the `lineinfile` module usage, please, use `ansible.builtin.copy` and the `to_nice_yaml` filter to properly create a YAML file within Ansible.